test/passes/legalize-js-interface_all-features.txt

```diff
@@ -2,16 +2,18 @@
  (type $none_=>_i32 (func (result i32)))
  (type $none_=>_i64 (func (result i64)))
  (type $i32_i32_i32_i32_i32_=>_none (func (param i32 i32 i32 i32 i32)))
+ (type $i32_i64_i64_=>_none (func (param i32 i64 i64)))
  (type $none_=>_none (func))
  (type $i32_=>_none (func (param i32)))
- (type $i32_i64_i64_=>_none (func (param i32 i64 i64)))
+ (import "env" "imported" (func $imported (result i64)))
+ (import "env" "other" (func $other (param i32 i64 i64)))
  (import "env" "ref-func-arg" (func $ref-func-arg (result i64)))
  (import "env" "setTempRet0" (func $setTempRet0 (param i32)))
  (import "env" "getTempRet0" (func $getTempRet0 (result i32)))
  (import "env" "imported" (func $legalimport$imported (result i32)))
  (import "env" "other" (func $legalimport$other (param i32 i32 i32 i32 i32)))
  (import "env" "ref-func-arg" (func $legalimport$ref-func-arg (result i32)))
- (elem declare func $ref-func-arg)
+ (elem declare func $legalfunc$ref-func-arg)
```

Comment:
What causes this change? And it is unrelated, but what is this `declare` keyword? (I checked reference types, TFR, and GC proposals but couldn't find it..)

Reply:
@aheejin A declared element segment is used as a forward declaration hint to engines so that they know what functions can be referenced using `ref.func`. It's completely unrelated to tables and other element segment types, and since they are not available at runtime, we don't represent them in the IR. Instead, we ignore this segment on parsing and regenerate it on writing or printing. Only functions that haven't been referenced in other element segments will be included in the declared segment, and only one segment of this type is emitted.

Reply:
I see. Thanks for the explanation!

Reply:
About the change: the change is because the `ref.func` lower down has been replaced to refer to another function. This has to correspond to that.